howto/stream/exchange-oob-data.md

```diff
@@ -2,127 +2,151 @@
-```
-
-The following instructions guide you to exchange OOB data using a specific implementation version:
-
-(sec-oob-data-version-2)=
-## Version 2: full-duplex bidirectional data transmission
+Enabling out-of-band (OOB) data transmission between an Android application and a WebRTC client makes it possible to exchange data and trigger actions between an Android application and a WebRTC client. Anbox Cloud provides a full-duplex bidirectional data transmission mode in which data can flow in both directions at the same time.
 
 The following instructions will walk you through how to set up data channels and perform data transmission in both directions between an Android application and a WebRTC platform.
 
-Before proceeding, ensure that you have set up a web-based streaming client using the Anbox Cloud streaming stack. If you haven't done so, please follow our setup [guide](https://documentation.ubuntu.com/anbox-cloud/en/latest/tutorial/stream-client/).
-
 ### Prepare your web application
 
-In your web-based client application, import the {ref}`sec-streaming-sdk`.
+This guide builds upon the [streaming client setup tutorial](https://documentation.ubuntu.com/anbox-cloud/en/latest/tutorial/stream-client/) for Anbox Cloud. Ensure you have completed the setup of a web-based streaming client as described in the tutorial till the [step where you set up stream client](https://documentation.ubuntu.com/anbox-cloud/en/latest/tutorial/stream-client/#implement-the-stream-client).
 
-Create a data channel (named `foo` in the following example) under the `dataChannels` property of an `AnboxStream` object and register event handlers that respond to the events sent from an Android application:
+#### Extend `AnboxStream` Configuration
+
+Now, to extend an `AnboxStream` object, add the `dataChannels` property to define a new data channel. For example:
 
 ```
-let stream = new AnboxStream({
-  ...
-  ...
-  dataChannels: {
-    "foo": {
-      callbacks: {
-        close: () => {
-          console.log('data channel is closed')
-        },
-        open: () => {
-          console.log('data channel is open')
-        },
-        error: (err) => {
-          console.log(`error: ${err}`)
-        },
-        message: (data) => {
-          console.log(`data received: ${data}`)
-        }
-      }
-    }
-  }
-});
+  <body>
+    <div id="anbox-stream"></div>
+    <div id="controls" style="position: absolute; top: 20px; left: 20px; z-index: 10">
+      <input type="text" id="textBox" placeholder="Send message">
+      <button id="sendButton">Send</button>
+    </div>
+
+    <script>
+      ...
+      window.onload = () => {
+        const stream = new AnboxStream({
+          targetElement: "anbox-stream",
+          ...
+          dataChannels: {
```

Comment:
Can we add a comment inline to explain / highlight that this is the relevant part of the code we want the user to notice?